src/requestManager.cpp

```diff
@@ -297,15 +297,17 @@ void CUnknownObj::AddSource(CNode* from)
       }
       CNodeRequestData req(from);
       for (ObjectSourceList::iterator i = availableFrom.begin(); i != availableFrom.end(); ++i)
-        {
-	  if (i->desirability < req.desirability)
-	    {
+      {
+          if (i->desirability < req.desirability)
+          {
               availableFrom.insert(i, req);
-              return;
-	    }
-        }
+              return true;
+           }
+      }
       availableFrom.push_back(req);
+      return true;
     }
```

Comment:
maybe a separate PR.  This PR just changes the function to return true/false to clean up some unnecessarily verbose logging